zerver/lib/markdown/__init__.py

```diff
@@ -2024,9 +2070,11 @@ def zulip_specific_link_changes(self, el: Element) -> None | Element:
         if not el.text or not el.text.strip():
             el.text = href
 
-        # Prevent linkifiers from running on the content of a Markdown link, breaking up the link.
-        # This is a monkey-patch, but it might be worth sending a version of this change upstream.
-        el.text = markdown.util.AtomicString(el.text)
+        # Prevent other linkifiers from running on the content of a Markdown link,
+        # which can result in nested links. We've removed this in favor of the
+        # `UnnestLinks` treeprocessor, which allows for other inline markdown
+        # to render (e.g. bold and italics) and cleans up any nested links later.
+        # el.text = markdown.util.AtomicString(el.text)
```

Comment:
We'll want to see tests that cover the type of corner cases that this previous code was meant to cover.